code/modules/admin/sound_emitter.dm

```diff
@@ -0,0 +1,149 @@
+#define SOUND_EMITTER_LOCAL "local" //Plays the sound like a normal heard sound
+#define SOUND_EMITTER_DIRECT "direct" //Plays the sound directly to hearers regardless of pressure/proximity/et cetera
+
+#define SOUND_EMITTER_RADIUS "radius" //Plays the sound to everyone in a radius
+#define SOUND_EMITTER_ZLEVEL "zlevel" //Plays the sound to everyone on the z-level
+#define SOUND_EMITTER_GLOBAL "global" //Plays the sound to everyone in the game world
+
+//Admin sound emitters with highly customizable functions!
+/obj/effect/sound_emitter
+	name = "sound emitter"
+	desc = "Emits sounds, presumably."
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "shield2"
+	invisibility = INVISIBILITY_OBSERVER
+	anchored = TRUE
+	density = FALSE
+	opacity = FALSE
+	alpha = 175
+	var/sound_file //The sound file the emitter plays
+	var/sound_volume = 50 //The volume the sound file is played at
+	var/play_radius = 3 //Any mobs within this many tiles will hear the sounds played if it's using the appropriate mode
+	var/motus_operandi = SOUND_EMITTER_LOCAL //The mode this sound emitter is using
+	var/emitter_range = SOUND_EMITTER_ZLEVEL //The range this emitter's sound is heard at; this isn't a number, but a string (see the defines above)
+
+/obj/effect/sound_emitter/Destroy(force)
+	if(!force)
+		return QDEL_HINT_LETMELIVE
+	. = ..()
+
+/obj/effect/sound_emitter/singularity_act()
+	return
+
+/obj/effect/sound_emitter/examine(mob/user)
+	..()
+	if(!isobserver(user))
+		return
+	to_chat(user, "<span class='boldnotice'>Sound File:</span> [sound_file ? sound_file : "None chosen"]")
+	to_chat(user, "<span class='boldnotice'>Mode:</span> [motus_operandi]</span>")
+	to_chat(user, "<span class='boldnotice'>Range:</span> [emitter_range]</span>")
+	to_chat(user, "<b>Sound is playing at [sound_volume]% volume.</b>")
+	if(user.client.holder)
```

Comment:
please add an explicit rights check, not every downstream wants all users with an admin holder to be able to access thigns like this.

Also it's lazy coding.

Reply:
I haven't messed with admin code enough to know that by heart, bear with me. I'll quickly do a check for sound rights.